### ISSUE TITLE
In experiments/train.py，training labels problem

### ISSUE THREAD

**Issue by Jean-ZW** [microsoft/KBLaM#79]
<img width="979" height="589" alt="Image" src="https://github.com/user-attachments/assets/c27ab759-c115-44f2-8983-6cddc4ea9871" />

As shown over here, the 2 labels creating method only masked the starting special tokens until the start of user content, not the whole query inputs until the start assistant content like what regular SFT does. 

Does this make sense? train not just the answer but the query content as well? Or what it wants is not only, just train for instruction following, answer generation ability, but also a bit like pretrain for latent sentiment space alignment for the adapters?

And what does these comment notes means? "Not sure... Possibly what they want is...", "...Not 100% this is correct" 
Aren't the authors of 《KBLaM》 provided this code for paper reproduction? What labels they used for training exactly? Does anyone have any ideas?

**Comment by xidulu** [comment]
The idea is to fine-tune only over the answer part, and that's what `_create_labels_for_llama` is about.

> Does this make sense? train not just the answer but the query content as well?

I didn't know why this does not make sense, I don't see why we need to sft over the answer to learn the adapter. 



> And what does these comment notes means?

Because I am the one the wrote the code, these comments are left by my colleagues trying to understand the meaning of the code.
